### PR TITLE
Fix URL `path` parsing

### PR DESCRIPTION
**Pull request by codenirvana**
Fixes postmanlabs/postman-app-support#4663

- Handle empty path issue
- Properly parse string path in JSON representation

### CODE REVIEW

**Comment by kamalaknn** [comment]
@codenirvana This is updating a very critical code path. Can you do some benchmark tests?

**Comment by codenirvana** [comment]
@kamalaknn Similar performance.

**Before:**
```
1000 String Input   : 54.990ms
String + JSON Input : 1.237ms
Path Edge Case      : 0.229ms
Very Large Input    : 1.407ms

# 100 Iterations : 25.45s
```

**After:**
```
1000 String Input   : 54.742ms
String + JSON Input : 1.221ms
Path Edge Case      : 0.176ms
Very Large Input    : 1.370ms

# 100 Iterations : 25.50s
```

File: lib/collection/url.js [hunk 1]
```diff
@@ -102,7 +102,12 @@ _.assign(Url.prototype, /** @lends Url.prototype */ {
         }
 
         // expand string path name
-        _.isString(path) && (path = path.split(PATH_SEPARATOR));
+        if (_.isString(path)) {
+            path && (path = path.replace(regexes.trimPath, MATCH_1)); // remove leading slash for valid path
+            // if path is blank string, we set it to undefined, if '/' then single blank string array
+            path = !path ? undefined : (path === PATH_SEPARATOR ? [E] : path.split(PATH_SEPARATOR));
```
Comment (kunagpal) [inline lib/collection/url.js]:
```js
path = path ? (path === PATH_SEPARATOR ? [E] : path.split(PATH_SEPARATOR)) : undefined
```

**Comment by kunagpal** [comment]
@shamasis I'm not quite sure, this looks like a job for `git blame` 😅
